### PR TITLE
Lua: Implement awesome.pixbuf_to_surface()

### PR DESCRIPTION
**Pull request by psychon**
This is a new function that was added to awesome in commit ee944f4da83.
It gets a pointer to a GdkPixbuf instance as its argument and returns a
cairo image surface with the same content. It works similar to the
existing awesome.load_image(), but without loading a pixbuf from a file,
thus it is more general.

Fixes: https://github.com/way-cooler/way-cooler/issues/545
Signed-off-by: Uli Schlachter <psychon@znc.in>

### CODE REVIEW

**Comment by psychon** [comment]
I forgot: I tested this with the following Lua code. Both resulting PNG files are equally broken, so this new code seems to work fine. I'll debug `load_surface_from_pixbuf` next...
```lua
local lgi = require("lgi")
local cairo = lgi.cairo
local GdkPixbuf = lgi.GdkPixbuf

local img = "/home/psychon/Wallpaper/flowers-on-black.png"

-- Old version, load_image
local s = cairo.Surface(awesome.load_image(img), true)
s:write_to_png("/tmp/method_old.png")

-- New version, pixbuf_to_surface
local pb = GdkPixbuf.Pixbuf.new_from_file(img)
local s = cairo.Surface(awesome.pixbuf_to_surface(pb._native), true)
s:write_to_png("/tmp/method_new.png")

print("done")
```
(Oh and: I only tested the old method in awesome, my currently running awesome instance is too old for `pixbuf_to_surface`)

**Comment by acrisci** [comment]
:+1: I can actually use the default config with this.

**Comment by Timidger** [comment]
Getting the raw pointer is an issue in aware of for using the Cairo library.

The way you solved it is my preferred way. Though I have not reached out to the Cairo-rs people on changing it.

File: src/awesome/awesome.rs [hunk 1]
```diff
@@ -248,6 +249,19 @@ fn load_image<'lua>(lua: &'lua Lua, file_path: String) -> rlua::Result<Value<'lu
     rlua::LightUserData(surface_ptr as _).to_lua(lua)
 }
 
+/// Convert a pixbuf to a cairo image surface.
+/// Returns either a cairo surface as light user data, nil and an error message
+fn pixbuf_to_surface<'lua>(lua: &'lua Lua, pixbuf: LightUserData) -> rlua::Result<Value<'lua>> {
```
Comment (Timidger) [inline src/awesome/awesome.rs]: The raw pointer being passed in here from Lua with no checks possible for validity makes me sad. But there's nothing we can do I suppose.

Perhaps once Way Cooler is stable we can look into making that safer.

**Comment by psychon** [comment]
> The way you solved it is my preferred way. Though I have not reached out to the Cairo-rs people on changing it.

That's just copy&paste from `load_image` (/me hides).
I saw that a `to_glib_none` exists and tried to use it, but  hit an `unimplemented!()` :-(

**Pull request merged**